main/src/cgeo/geocaching/EditWaypointActivity.java

```diff
@@ -539,6 +539,9 @@ protected Void doInBackground(final Void... params) {
                     finishHandler.sendEmptyMessage(SAVE_ERROR);
                     return null;
                 }
+                if (own) {
+                    waypoint.setPrefix(waypoint.getPrefix()+String.valueOf(cache.getWaypoints().size()));
```

Comment:
Seems I never finished that comment 4 days ago:
Here I think the prefix is not guaranteed to be unique. If you first add two waypoints, than delete the older one and add another, I assume you end up with a duplicate prefix again.